src/app/CommandHandlerInterface.h

```diff
@@ -239,13 +228,27 @@ class CommandHandlerInterface
      *  void Func(HandlerContext &handlerContext, const RequestT &requestPayload);
      */
     template <typename RequestT, typename FuncT, typename std::enable_if_t<DataModel::IsFabricScoped<RequestT>::value, bool> = true>
-    void HandleCommand(HandlerContext & handlerContext, FuncT func)
+    __attribute__((always_inline)) inline void HandleCommand(HandlerContext & handlerContext, FuncT func)
     {
-        if (!handlerContext.mCommandHandled && (handlerContext.mRequestPath.mClusterId == RequestT::GetClusterId()) &&
-            (handlerContext.mRequestPath.mCommandId == RequestT::GetCommandId()))
+        if (!PrehandleCommand(handlerContext, RequestT::GetClusterId(), RequestT::GetCommandId()))
         {
-            RequestT requestPayload;
+            return;
+        }
+        RequestT requestPayload;
+        if (!HandleDecode(handlerContext,
+                          requestPayload.Decode(handlerContext.mPayload, handlerContext.mCommandHandler.GetAccessingFabricIndex())))
+        {
+            return;
+        }
 
+        func(handlerContext, requestPayload);
+    }
+
+    inline bool PrehandleCommand(HandlerContext & handlerContext, ClusterId expectedClusterId, CommandId expectedCommandId)
```

Comment:
Nit: please describe what "prehandle" means, especially what true/false means, via some documentation or renaming it to a more obvious name.